### PR TITLE
merge joe/syncvss

### PR DESCRIPTION
**Pull request by A-Manning**

### CODE REVIEW

File: src/dkg.rs [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+        self.vss
+            .iter()
+            .filter_map(|(_, vss)| {
+                if let VSSState::Success { final_secret } = vss.state {
+                    Some(final_secret)
+                } else {
+                    None
+                }
+            })
+            // .sum() //TODO: it would be nice to use Sum trait here, but not implemented for G1Affine
```
Comment (ggkitsas) [inline src/dkg.rs:119]: created an arkworks issue, might grab it
https://github.com/arkworks-rs/algebra/issues/288

File: src/fastkzg.rs [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+        skip_leading_zeros_and_convert_to_bigints(polynomial);
+
+    let commitment = VariableBaseMSM::multi_scalar_mul(
+        &powers_of_g[num_leading_zeros..],
+        &plain_coeffs,
+    );
+    Ok(commitment.into())
+}
+
+pub fn setup<R: rand::RngCore>(
```
Comment (ggkitsas) [inline src/fastkzg.rs:266]: shall I replace this function with the trusted setup loading function?

File: src/msg.rs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+pub struct SignedMessage {
+    msg_bytes: Vec<u8>,
+    signature: ed25519::Signature,
+    pub signer: ed25519::PublicKey,
+}
+
+#[test]
+fn test_ark_serde() {
+    use ark_bls12_381::G1Affine;
+    //use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
```
Comment (ggkitsas) [inline src/msg.rs:79]:
```suggestion
```

File: src/msg.rs [hunk 4]
```diff
@@ -0,0 +1,92 @@
+use ed25519_dalek as ed25519;
+//use ed25519_dalek::Signature;
```
Comment (ggkitsas) [inline src/msg.rs:2]:
```suggestion
```

File: src/msg.rs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+use ed25519_dalek::Signer;
+
+use crate::syncvss;
+//use ark_bls12_381::G1Affine;
```
Comment (ggkitsas) [inline src/msg.rs:6]:
```suggestion
```

**Review by ggkitsas — COMMENTED** [review]
LGTM

**Comment by joebebel** [comment]
changes committed via other branch

**Pull request closed without merging**